### PR TITLE
[16.0][MIG] rma_account

### PR DESCRIPTION
**Pull request by DavidJForgeFlow**
Migration to 16.0

@ForgeFlow

### CODE REVIEW

File: rma_account/models/account_move.py [hunk 1]
```diff
@@ -50,8 +50,6 @@ def _post_process_invoice_line_from_rma_line(self, new_line, rma_line):
             new_line._get_computed_name(),
```
Comment (florian-dacosta) [inline rma_account/models/account_move.py]: Replace new_line.name, since it becamse a computed field.

File: rma_account/models/account_move.py [hunk 2]
```diff
@@ -50,8 +50,6 @@ def _post_process_invoice_line_from_rma_line(self, new_line, rma_line):
...
         )
         new_line.account_id = new_line._get_computed_account()
```
Comment (florian-dacosta) [inline rma_account/models/account_move.py]: seems useless, as account_id became a computed field.

File: rma_account/models/account_move.py [hunk 3]
```diff
@@ -272,3 +269,40 @@ def _stock_account_get_anglo_saxon_price_unit(self):
                     self.company_id
                 )._compute_price(average_price_unit, self.product_uom_id)
         return price_unit
+
+    def _get_computed_name(self):
```
Comment (florian-dacosta) [inline rma_account/models/account_move.py]: Why is this method needed, that is basically what the compute method of the name field does right ?
https://github.com/odoo/odoo/blob/16.0/addons/account/models/account_move_line.py#L460

File: rma_account/models/account_move.py [hunk 4]
```diff
@@ -272,3 +269,40 @@ def _stock_account_get_anglo_saxon_price_unit(self):
...
+        if self.journal_id.type == "sale":
+            if product.description_sale:
+                values.append(product.description_sale)
+        elif self.journal_id.type == "purchase":
+            if product.description_purchase:
+                values.append(product.description_purchase)
+        return "\n".join(values)
+
+    def _get_computed_account(self):
+        self.ensure_one()
```
Comment (florian-dacosta) [inline rma_account/models/account_move.py]: Same here, not sure it is needed, it is already computed here : https://github.com/odoo/odoo/blob/16.0/addons/account/models/account_move_line.py#L487

File: rma_account/models/rma_order_line.py [hunk 5]
```diff
@@ -82,7 +82,6 @@ def _compute_refund_count(self):
         string="Refund Lines",
         domain=[
             ("move_id.move_type", "in", ["in_refund", "out_refund"]),
-            ("exclude_from_invoice_tab", "=", False),
```
Comment (florian-dacosta) [inline rma_account/models/rma_order_line.py]: Replace by `("display_type", "=", "product"),` to keep the same behavior ?

File: rma_account/models/rma_order_line.py [hunk 6]
```diff
@@ -129,21 +128,17 @@ def _onchange_product_id(self):
             res["domain"] = {}
         if not self.product_id:
             domain = [
-                "&",
                 "|",
                 ("move_id.partner_id", "=", self.partner_id.id),
                 ("move_id.partner_id", "child_of", self.partner_id.id),
-                ("exclude_from_invoice_tab", "=", False),
```
Comment (florian-dacosta) [inline rma_account/models/rma_order_line.py]: Replace by `("display_type", "=", "product"),` to keep the same behavior ?

File: rma_account/models/rma_order_line.py [hunk 7]
```diff
@@ -129,21 +128,17 @@ def _onchange_product_id(self):
...
             ]
             res["domain"]["account_move_line_id"] = domain
         else:
             domain = [
-                "&",
                 "&",
                 "|",
                 ("move_id.partner_id", "=", self.partner_id.id),
                 ("move_id.partner_id", "child_of", self.partner_id.id),
-                ("exclude_from_invoice_tab", "=", False),
```
Comment (florian-dacosta) [inline rma_account/models/rma_order_line.py]: Replace by `("display_type", "=", "product"),` to keep the same behavior ?

File: rma_account/wizards/rma_add_account_move.py [hunk 8]
```diff
@@ -122,9 +122,7 @@ def _get_existing_invoice_lines(self):
     def add_lines(self):
         rma_line_obj = self.env["rma.order.line"]
         existing_invoice_lines = self._get_existing_invoice_lines()
-        for line in self.line_ids.filtered(
-            lambda aml: aml.exclude_from_invoice_tab is False
-        ):
+        for line in self.line_ids.filtered(lambda l: l.product_id.id):
```
Comment (florian-dacosta) [inline rma_account/wizards/rma_add_account_move.py]: `for line in self.line_ids.filtered(lambda aml: aml.display_type == "product"):` seems more reliable to me

**Review by florian-dacosta — COMMENTED** [review]
Thanks for the migration. Some remarks, I am not sure to understand all the changes

File: rma_account/models/account_move.py [hunk 9]
```diff
@@ -0,0 +1,272 @@
...
+            invl.used_in_rma_line_count = len(rma_lines)
+
+    def _compute_rma_count(self):
+        for invl in self:
+            rma_lines = invl.mapped("rma_line_id")
+            invl.rma_line_count = len(rma_lines)
+
+    used_in_rma_line_count = fields.Integer(
+        compute="_compute_used_in_rma_line_count", string="# of RMA"
+    )
```
Comment (OriolMForgeFlow) [inline rma_account/models/account_move.py]: The function `_compute_used_in_rma_line_count` is not defined. Perhaps you should refer to `_compute_used_in_rma_count`.

File: rma_account/models/account_move.py (hunk 1 above)
Comment (DavidJForgeFlow) [reply rma_account/models/account_move.py]: Done, thanks @florian-dacosta!

File: rma_account/models/account_move.py (hunk 2 above)
Comment (DavidJForgeFlow) [reply rma_account/models/account_move.py]: Done!

File: rma_account/models/account_move.py (hunk 3 above)
Comment (DavidJForgeFlow) [reply rma_account/models/account_move.py]: Done!

File: rma_account/models/account_move.py (hunk 4 above)
Comment (DavidJForgeFlow) [reply rma_account/models/account_move.py]: Done!

File: rma_account/models/rma_order_line.py (hunk 5 above)
Comment (DavidJForgeFlow) [reply rma_account/models/rma_order_line.py]: Done!

File: rma_account/models/rma_order_line.py (hunk 6 above)
Comment (DavidJForgeFlow) [reply rma_account/models/rma_order_line.py]: Done!

File: rma_account/models/rma_order_line.py (hunk 7 above)
Comment (DavidJForgeFlow) [reply rma_account/models/rma_order_line.py]: Done!

File: rma_account/wizards/rma_add_account_move.py (hunk 8 above)
Comment (DavidJForgeFlow) [reply rma_account/wizards/rma_add_account_move.py]: Done!

File: rma_account/models/account_move.py (hunk 9 above)
Comment (DavidJForgeFlow) [reply rma_account/models/account_move.py]: Done, thanks @OriolMForgeFlow !

**Pull request merged**